plugins/in_kafka/in_kafka.c

```diff
@@ -246,6 +248,22 @@ static int in_kafka_init(struct flb_input_instance *ins,
         goto init_error;
     }
 
+    conf = flb_input_get_property("data_format", ins);
```

Comment:
This configuration mechanism has been deprecated, add the fields offset to the config_map structure and use it directly.

Reply:
I'm aware of that, however it is not clear how we can handle "enum" properties using this mechanism. The data_format property currently supports 2 values, "none" and "json", but will likely support other values such as "avro" in the future.

Using the field offset, how can I check that the value set matches one of the valid strings (also ignoring case)?

Reply:
We set it up as a string and then process that in the configuration function and set a different field with the numeric value.